openerp/models.py

```diff
@@ -4890,6 +4890,24 @@ def _check_recursion(self, cr, uid, ids, context=None, parent=None):
                     return False
         return True
 
+    # _constraint method making all tag a unique case and accent insensitive
+    @api.v7
+    def _check_unique(self, cr, uid, ids, context=None):
+        rec = self.browse(cr, uid, ids, context=None)
+        try:
+            return rec._check_unique_accent()
+        except ValidationError:
+            return False
+    @api.one
+    @api.v8
+    def _check_unique_accent(self):
+        unaccent = expression.get_unaccent_wrapper(self._cr)
+        query = ("""SELECT id from {table_name} WHERE lower({name}) = lower({tag}) and id != {current_id} LIMIT 1""".format(table_name=str(self._table), name=unaccent('name'), tag=unaccent('%s'), current_id=('%s')))
+        self._cr.execute(query, [self.name, self.id])
+        if self._cr.fetchall():
```

Comment:
Yes it could be faster, if we use search_count() but here we used unaccent function of PostgreSQL on column "name", so we can not able to apply this type of domain in search_count() that is why here we execute query and fetch result.

And about self._cr.fetchall() can we use self._cr.rowcount ? Rowcount will return the number of rows in the query.